packages/cli/src/lib/generators/CodeGenerator.ts

```diff
@@ -0,0 +1,104 @@
+import { BuildConfig } from "../Compiler";
+import { SchemaComposer } from "../SchemaComposer";
+
+import chalk from "chalk";
+import fs, { readFileSync } from "fs";
+import path from "path";
+import Mustache from "mustache";
+import {
+  OutputDirectory,
+  OutputEntry,
+  writeDirectory,
+} from "@web3api/schema-bind";
+import { TypeInfo, parseSchema } from "@web3api/schema-parse";
+
+export class CodeGenerator {
+  private _schema: string | undefined = "";
+  constructor(private _templateFile: string, private _config: BuildConfig) {}
+
+  public async generateCode(): Promise<boolean> {
+    // Make sure that the output dir exists, if not create a new one
+    if (!fs.existsSync(this._config.outputDir)) {
+      fs.mkdirSync(this._config.outputDir);
+    }
+
+    // Compose schema from manifest
+    const schemaComposer = new SchemaComposer(this._config);
+    const manifest = await schemaComposer.loadManifest();
+    const composedSchema = await schemaComposer.composeSchemas(manifest);
+    const typeInfo = parseSchema(composedSchema.combined!);
```

Comment:
Why the `!`?

Reply:
Because `parseSchema` doesn't allow null values. 😣

Reply:
I think we should add an if check here, and throw an error if combined is falsey. Also it might be worth revisiting the schema/compose/src/index.ts file to see why combined can be undefined, is that an option of that function? Might make sense to just make it always return a string and not string | undefined.